app/src/core/components/SubmitButton.tsx

```diff
@@ -27,7 +27,9 @@ export default function SubmitButton({
     type = 'submit',
     ...rest
 }: PropsWithChildren<SubmitButtonProps>) {
-    const [loading, setLoading] = useState(isLoading ?? false);
+    const [isSubmitResolving, setIsSubmitResolving] = useState(false);
+    const isButtonLoading = isSubmitResolving || isLoading;
```

Comment:
@bakerac4 this is necessary because by passing it directly into `useState()`, the `isLoading` prop was only setting the default starting value for the `loading` state on its initial render. Updating the prop value outside of the button would not cause it to toggle the loading state as we'd want, as `useState` does not pull in its default value on subsequent re-renders.